src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            grow(size + 1);
```

Comment:
Don`t sent size its redundant, just create method check capacity or something else

Reply:
https://github.com/mate-academy/jv-array-list/pull/1200#discussion_r1374809713
I believe that in this place you need to pass the minimum size to which you need to increase the array, otherwise it increases the execution time.

Reply:
The main thing of resizing that it check size and grow size automatically, you don`t need manage it by yourself

Reply:
If you want enhancement performance of addAll you can make it only in that method without reusing method add